### DISCUSSION TITLE
A put method can't followed by another put method. Error: Unprocessable Entity

### DISCUSSION THREAD

**Discussion by shizidushu** [fastapi/fastapi#7524] [Questions] [answered]
### Describe the bug

Two consecutive put method in a file, the second one doesn't work. If I didn't miss something, it would be a bug.

### To Reproduce

1. Create a file with:

```Python
from fastapi import FastAPI, Body

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}

@app.put("/{user_id}")
def update_user(
    *,
    user_id: int,
):
    return user_id

@app.put("/update_role")
def update_role_user(
    *,
    username: str = Body(...),
    role_name: str = Body(...)):
    return username
```

3. Open the browser and go the http://127.0.0.1:8000/docs
4. Try and execute with default parameters. `curl -X PUT "http://127.0.0.1:8000/update_role" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"username\":\"string\",\"role_name\":\"string\"}"`
5. Return code 422. The response body
```
{
  "detail": [
    {
      "loc": [
        "path",
        "user_id"
      ],
      "msg": "value is not a valid integer",
      "type": "type_error.integer"
    }
  ]
}
```

### Environment

- OS:  Windows 10
- FastAPI Version 0.54.1
- Python 3.8.2

**Comment by phy25** [comment, marked as answer]
https://www.starlette.io/routing/#route-priority

TLDR: Order counts, or just do a `@app.put("/{user_id:int}")`.

**Comment by shizidushu** [comment]
@phy25 Thanks for your reply. Just read the doc. The first `put` route is more general so the request just go to it.

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 ! :cake: :bow: 

Thanks for reporting back and closing the issue @shizidushu  :+1:
